src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -968,35 +1074,38 @@ private static FunctionMetadata ParseFunctionMetadata(string functionName, JObje
             return functionMetadata;
         }
 
-        public static Collection<FunctionMetadata> ReadFunctionMetadata(ScriptHostConfiguration config, TraceWriter traceWriter, ILogger logger, Dictionary<string, Collection<string>> functionErrors, ScriptSettingsManager settingsManager = null)
+        public static Collection<FunctionMetadata> ReadFunctionMetadata(IEnumerable<string> functionDirectories, TraceWriter traceWriter, ILogger logger, Dictionary<string, Collection<string>> functionErrors, ScriptSettingsManager settingsManager = null, IEnumerable<string> functionWhitelist = null)
         {
             var functions = new Collection<FunctionMetadata>();
             settingsManager = settingsManager ?? ScriptSettingsManager.Instance;
 
-            if (config.Functions != null)
+            if (functionWhitelist != null)
             {
-                string msg = $"A function whitelist has been specified, excluding all but the following functions: [{string.Join(", ", config.Functions)}]";
+                string msg = $"A function whitelist has been specified, excluding all but the following functions: [{string.Join(", ", functionWhitelist)}]";
                 traceWriter.Info(msg);
             }
 
-            foreach (var scriptDir in Directory.EnumerateDirectories(config.RootScriptPath))
+            foreach (var scriptDir in functionDirectories)
             {
                 string functionName = null;
-
                 try
                 {
                     // read the function config
                     string functionConfigPath = Path.Combine(scriptDir, ScriptConstants.FunctionMetadataFileName);
-                    if (!File.Exists(functionConfigPath))
```

Comment:
Saved another file operation here, by optimistically attempting to read the file w/o the existence check